File: maven-model-reader-plugin/src/main/java/ru/rzn/gmyasoedov/model/reader/plugins/GroovyMavenPlugin.java
package ru.rzn.gmyasoedov.model.reader.plugins;

import org.apache.maven.model.Plugin;
import org.apache.maven.project.MavenProject;

public class GroovyMavenPlugin implements PluginProcessor{
    @Override
    public String groupId() {
        return "org.codehaus.gmaven";
    }

    @Override
    public String artifactId() {
        return "groovy-maven-plugin";
    }

    @Override
    public void process(MavenProject project, Plugin plugin) {
        AbstractGroovyPluginProcessor.process(project, plugin);
    }
}

File: gmaven/src/main/kotlin/ru/rzn/gmyasoedov/gmaven/extensionpoints/plugin/GroovyMavenPlugin.kt
package ru.rzn.gmyasoedov.gmaven.extensionpoints.plugin

class GroovyMavenPlugin : MavenFullImportPlugin {
    override fun getGroupId() = "org.codehaus.gmaven"

    override fun getArtifactId() = "groovy-maven-plugin"
}